### PR TITLE
server: class, not mixin

### PR DESCRIPTION
**Pull request by mxmCherry**
As suggested in https://github.com/bsm/grpcx/issues/4

### CODE REVIEW

File: grpcx.gemspec [hunk 1]
```diff
@@ -15,6 +15,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'activesupport', '>= 5.0'
   s.add_dependency 'grpc', '>= 1.8.0'
+  s.add_dependency 'grpclb', '~> 0.4.4'
```
Comment (mxmCherry) [inline grpcx.gemspec:18]: Please, double check version constraint... It's not `1.x.x` yet, so probs even worth making a single-version constraint.

File: lib/grpcx/server/interceptors/active_record.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
+require 'grpclb/server'
+
 module Grpcx
-  module Server
+  class Server < Grpclb::Server
```
Comment (mxmCherry) [inline lib/grpcx/server/interceptors/active_record.rb:4]: I dislike this, but seems no way round it, except, probably, changing naming scheme to smth like `Grpcx::Interceptors::Server::ActiveRecord`

File: lib/grpcx/server/interceptors/instrumentation.rb [hunk 3]
```diff
@@ -1,8 +1,12 @@
 require 'active_support/notifications'
+require 'grpclb/server'
 
 module Grpcx
-  module Server
+  class Server < Grpclb::Server
```
Comment (mxmCherry) [inline lib/grpcx/server/interceptors/instrumentation.rb:5]: I dislike this, but seems no way round it, except, probably, changing naming scheme to smth like `Grpcx::Interceptors::Server::Instrumentation`

File: lib/grpcx/server/interceptors/rescue.rb [hunk 4]
```diff
@@ -1,7 +1,11 @@
+require 'grpclb/server'
+
 module Grpcx
-  module Server
+  class Server < Grpclb::Server
```
Comment (mxmCherry) [inline lib/grpcx/server/interceptors/rescue.rb:4]: I dislike this, but seems no way round it, except, probably, changing naming scheme to smth like `Grpcx::Interceptors::Server::Rescue`

**Review by mxmCherry — COMMENTED** [review]
There are some changes that I don't like, like interceptor modules - it has to reference `class Grpcx::Server < Grpclb::Server`. Please, pay extra attention to these.

File: lib/grpcx/server.rb [hunk 5]
```diff
@@ -1,32 +1,27 @@
...
 require 'grpc/health/checker'
-require 'active_support/concern'
+require 'grpclb/server'
 require 'active_support/rescuable'
 require 'grpcx/server/interceptors'
 
 module Grpcx
-  module Server
-    extend ActiveSupport::Concern
+  class Server < Grpclb::Server
```
Comment (dim) [inline lib/grpcx/server.rb:8]: no, I don't like that. can we please remove `grpclb` again and instead do this:

1. rename `module Grpcx::Server` to `module Grpcx::Server::Mixin`
2. make `class Server < GRPC::RpcServer; include Mixin; ... `

Then we can still do:

```
class MyServer < Grpclb::Server
  include Grpcx::Server::Mixin
  ...
end
```

We should then do the same on grpclb side

File: lib/grpcx/server/interceptors/active_record.rb (hunk 2 above)
Comment (dim) [reply lib/grpcx/server/interceptors/active_record.rb:4]: pls revert

File: grpcx.gemspec (hunk 1 above)
Comment (dim) [reply grpcx.gemspec:18]: please remove, I don't want to embed this

**Review by dim — CHANGES_REQUESTED** [review]

File: lib/grpcx/server.rb (hunk 5 above)
Comment (mxmCherry) [reply lib/grpcx/server.rb:8]: > We should then do the same on grpclb side

You mean, make it mixin?

**Comment by mxmCherry** [comment]
IMO, this PR is pointless then. If we're keeping a mixin, and not using grpclb here - let's just leave it as-is, there's no benefit of doing the `Grpcx::Server` class at all. It really adds no value then.

**Pull request closed without merging**